crypto/keys.go

```diff
@@ -95,14 +102,25 @@ type ed25519PublicKey struct {
 	key ed25519.PublicKey
 }
 
+// secp256r1PublicKey wraps ecdsa.PublicKey to implement PublicKey interface
+type secp256r1PublicKey struct {
+	key             *ecdsa.PublicKey
+	compressedBytes []byte
+}
+
 // NewPrivateKey creates a new private key of a specific type based on the input
-func NewPrivateKey[T *secp256k1.PrivateKey | ed25519.PrivateKey](key T) PrivateKey {
+func NewPrivateKey[T *secp256k1.PrivateKey | *ecdsa.PrivateKey | ed25519.PrivateKey](key T) PrivateKey {
 	switch k := any(key).(type) {
 	case *secp256k1.PrivateKey:
 		if k == nil {
 			return nil
 		}
 		return &secp256k1PrivateKey{key: k}
+	case *ecdsa.PrivateKey:
+		if k == nil {
+			return nil
+		}
+		return nil
```

Comment:
todo: Please add a comment stating that private key is not supported for this type. 

todo: Please just return `nil` without the unnecessary equality check.